src/core/api/coap_api.cpp

```diff
@@ -218,9 +246,17 @@ otError otCoapSendRequestWithParameters(otInstance *              aInstance,
         VerifyOrExit(txParameters.IsValid(), error = OT_ERROR_INVALID_ARGS);
     }
 
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+
+    error = instance.GetApplicationCoap().SendMessage(*static_cast<Coap::Message *>(aMessage),
+                                                      *static_cast<const Ip6::MessageInfo *>(aMessageInfo),
+                                                      txParameters, aHandler, aContext, NULL, NULL);
+#else  // OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+
     error = instance.GetApplicationCoap().SendMessage(*static_cast<Coap::Message *>(aMessage),
                                                       *static_cast<const Ip6::MessageInfo *>(aMessageInfo),
                                                       txParameters, aHandler, aContext);
+#endif // OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
```

Comment:
Default parameters should avoid the need to special-case this.
```suggestion
    error = instance.GetApplicationCoap().SendMessage(*static_cast<Coap::Message *>(aMessage),
                                                      *static_cast<const Ip6::MessageInfo *>(aMessageInfo),
                                                      txParameters, aHandler, aContext);
```